compiler/rustc_hir_analysis/src/check/intrinsicck.rs

```diff
@@ -62,6 +62,7 @@ impl<'a, 'tcx> InlineAsmCtxt<'a, 'tcx> {
             ty::Int(IntTy::I64) | ty::Uint(UintTy::U64) => Some(InlineAsmType::I64),
             ty::Int(IntTy::I128) | ty::Uint(UintTy::U128) => Some(InlineAsmType::I128),
             ty::Int(IntTy::Isize) | ty::Uint(UintTy::Usize) => Some(asm_ty_isize),
+            ty::Float(FloatTy::F16) => Some(InlineAsmType::F16),
```

Comment:
Don't we have the same issues for `f128`?

Reply:
It's probably okay to handle f128 in a separate PR, since we might want a special test that it actually works right on s390x (it has f128 arithmetic operations but it looks like the float ABI might not use vector registers?). Still no harm in having it here without enabling it on a target of course